sonar-db/src/test/java/org/sonar/db/version/v60/FixProjectUuidOfDevelopersTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version.v60;
+
+import java.sql.SQLException;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FixProjectUuidOfDevelopersTest {
+
+  private static final String TABLE_PROJECTS = "projects";
+  private static final String PROJECT_UUID = "U1";
+  private static final String FILE_UUID = "U2";
+  private static final String DEVELOPER_UUID = "U3";
+  private static final String DEV1_IN_PROJECT_UUID = "U4";
+  private static final String DEV2_IN_PROJECT_UUID = "U5";
+
+  @Rule
+  public DbTester db = DbTester.createForSchema(System2.INSTANCE, FixProjectUuidOfDevelopersTest.class,
+    "in_progress_projects.sql");
+
+  private FixProjectUuidOfDevelopers underTest = new FixProjectUuidOfDevelopers(db.database());
+
+  @Test
+  public void migration_has_no_effect_on_empty_tables() throws SQLException {
+    underTest.execute();
+
+    assertThat(db.countRowsOfTable(TABLE_PROJECTS)).isEqualTo(0);
+  }
+
+  @Test
+  public void migration_fixes_project_uuid_of_rows_with_qualifier_DEV_PRJ() throws SQLException {
+    insertComponents();
+
+    underTest.execute();
+
+    verifyComponents();
+  }
+
+  @Test
+  public void migration_is_reentrant() throws SQLException {
+    insertComponents();
+
+    underTest.execute();
+    verifyComponents();
+
+    underTest.execute();
```

Comment:
this is how this test has been coded for all such migrations in 6.0 but my comment above makes me realize this test is missing some test to enforce no update has be made to the DB